browser/resources/brave_new_tab_page_refresh/components/search/search_results.tsx

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Button from '@brave/leo/react/button'
+import Icon from '@brave/leo/react/icon'
+
+import { SearchResultMatch, ClickEvent } from '../../models/search'
+import { useLocale } from '../context/locale_context'
+import { useAppActions, useAppState } from '../context/app_model_context'
+import { placeholderImageSrc } from '../../lib/image_loader'
+import { faviconURL } from '../../lib/favicon_url'
+import { Optional } from '../../lib/optional'
+import { SafeImage } from '../common/safe_image'
+
+import { style } from './search_results.style'
+
+function MatchImage(props: { match: SearchResultMatch }) {
+  const { getString } = useLocale()
+  const { imageUrl, iconUrl } = props.match
+  if (props.match.description === getString('searchAskLeoDescription')) {
+    return <Icon name='product-brave-leo' className='brave-leo-icon' />
+  }
+  if (!imageUrl) {
+    return <img className='icon' src={iconUrl || placeholderImageSrc} />
+  }
+  if (imageUrl.startsWith('chrome:')) {
+    return <img src={imageUrl} />
+  }
+  return <SafeImage src={imageUrl} />
+}
+
+interface URLResultOption {
+  kind: 'url'
+  url: string
+}
+
+interface MatchResultOption {
+  kind: 'match'
+  matchIndex: number
+  match: SearchResultMatch
+}
+
+export type ResultOption = URLResultOption | MatchResultOption
+
+interface Props {
+  options: ResultOption[]
+  selectedOption: Optional<number>
+  onOptionClick: (option: ResultOption, event: ClickEvent) => void
+  onSearchSuggestionsEnabled: () => void
+}
+
+export function SearchResults(props: Props) {
+  const { selectedOption, options } = props
+
+  const { getString } = useLocale()
+  const actions = useAppActions()
+
+  const searchSuggestionsEnabled =
+      useAppState((s) => s.searchSuggestionsEnabled)
+  const searchSuggestionsPromptDismissed =
+      useAppState((s) => s.searchSuggestionsPromptDismissed)
+
+  if (options.length === 0) {
+    return null
+  }
+
+  function renderSearchSuggestionsPrompt() {
+    if (searchSuggestionsEnabled || searchSuggestionsPromptDismissed) {
+      return null
+    }
+    return (
+      <div className='suggestions-prompt'>
+        <h4>{getString('searchSuggestionsPromptTitle')}</h4>
+        <p>
+          {getString('searchSuggestionsPromptText')}
+        </p>
+        <div className='actions'>
+          <Button
+            onClick={() => {
+              actions.setSearchSuggestionsEnabled(true)
+              props.onSearchSuggestionsEnabled()
```

Comment:
Hey I don't think this is what we want to do here. We should avoid needing to call an `onXXX` method when we call an action. Whatever is interested in the change should be listening to the current value of `searchSuggestionsEnabled`, and should be notified (probably over the mojom interface, but possibly just in the model).